packages/algorand-js/src/interpreter/ssc.ts

```diff
@@ -0,0 +1,150 @@
+  const appDelta = interpreter.globalApps.get(appId);
+  if (!appDelta) {
+    throw new TealError(ERRORS.TEAL.APP_NOT_FOUND, {
+      appId: appId
+    });
+  }
+  const globalState = appDelta["global-state"];
+
+  const keyValue = globalState.find(schema => compareArray(schema.key as Uint8Array, key));
+  const value = keyValue?.value;
+  if (value) {
+    return value?.bytes as Uint8Array || BigInt(value?.uint);
+  }
+  return undefined;
+}
+
+/**
+ * Description: add new key-value pair or updating pair with existing key in account
+ * for application id: appId, throw error otherwise
+ * @param appId: current application id
+ * @param account: account having local state data
+ * @param key: key to fetch value of from local state
+ * @param value: key to fetch value of from local state
+ */
+export function updateLocalState (appId: number, account: AccountState,
+  key: Uint8Array, value: StackElem): AppLocalState[] {
+  const localState = account["apps-local-state"];
+  const data = getKeyValPair(key, value); // key value pair to put
+
+  for (const l of localState) {
+    if (l.id === appId) { // find appId
+      const localDelta = l["key-value"];
+      const idx = localDelta.findIndex(schema => compareArray(schema.key as Uint8Array, key));
+
+      if (idx === -1) {
+        localDelta.push(data); // push new pair if key not found
+      } else {
+        localDelta[idx].value = data.value; // update value if key found
+      }
+      l["key-value"] = localDelta; // save updated state
+
+      assertValidSchema(l["key-value"], l.schema); // verify if updated schema is valid by config
+      return localState;
+    }
+  }
+
+  throw new TealError(ERRORS.TEAL.APP_NOT_FOUND, {
+    appId: appId
+  });
+}
+
+/**
+ * Description: add new key-value pair or updating pair with existing key in
+ * app's global data for application id: appId, throw error otherwise
+ * @param appId: current application id
+ * @param key: key to fetch value of from local state
+ * @param value: key to fetch value of from local state
+ * @param interpreter: interpreter object
+ */
+export function updateGlobalState (appId: number, key: Uint8Array,
+  value: StackElem, interpreter: Interpreter): SSCStateSchema[] {
+  const appDelta = interpreter.globalApps.get(appId);
+  if (!appDelta) {
+    throw new TealError(ERRORS.TEAL.APP_NOT_FOUND, {
+      appId: appId
+    });
+  }
+
+  const globalState = appDelta["global-state"];
+  const data = getKeyValPair(key, value); // key value pair to put
+  const idx = globalState.findIndex(schema => compareArray(schema.key as Uint8Array, key));
+  if (idx === -1) {
+    globalState.push(data); // push new pair if key not found
+  } else {
+    globalState[idx].value = data.value; // update value if key found
+  }
+  appDelta["global-state"] = globalState; // save updated state
+
+  assertValidSchema(appDelta["global-state"], appDelta["global-state-schema"]); // verify if updated schema is valid by config
+  return globalState;
+}
```

Comment:
this should be a method of a runtime. You can do this later, once you will create runtime -> in that case please add TODO in the file, and add a subtask to the Runtime PoC pivotal task.

Reply:
ok